### PR TITLE
added timeout on phatomjs 'exit' event

### PR DESCRIPTION
**Pull request by thegiantbeast**
Wait X milliseconds before processing the ‘exit’ event as ‘data event
might still be processing

### CODE REVIEW

**Comment by thegiantbeast** [comment]
**Issue:**
When running html-pdf module on parallel (required for some use cases) after a couple of processes, the 'exit' event is fired but the content is not yet fully ready to continue the process.

**Fix:**
Add a (configurable) timeout (60ms as default) on on the 'exit' event callback so it will wait before considering the phatomjs process 'closed'.

The same way that the official phantomjs handles it - [see here](https://github.com/Medium/phantomjs/blob/master/bin/phantomjs)

**Comment by thegiantbeast** [comment]
@marcbachmann

File: src/pdf.coffee [hunk 1]
```diff
@@ -90,17 +91,21 @@ module.exports = class PDF
...
-        return callback(err)
-      else
-        try
-          data = Buffer.concat(stdout).toString()?.trim()
-          data = JSON.parse(data)
-        catch err
+      # Wait X milliseconds after the 'exit' event being fired
+      # as 'data' event might still be processing
+      phantomExitTimeout = (options.phantomExitTimeout || 60)
+      setTimeout ->
```
Comment (marcbachmann) [inline src/pdf.coffee]: I'm sorry. In the link you provided they work around the problem that `console.log()` writes to stdout in the next tick. We don't have that issue in here. You probably ran into an a timing issue in the phantomjs process. What's your use case?

**Comment by marcbachmann** [comment]
Won't merge this as it's not really a fix for the problem.
We have to solve that properly.

**Pull request closed without merging**